include/aspect/fluid_pressure_boundary_conditions/density.h

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__fluid_pressure_boundary_conditions_density_h
+#define __aspect__fluid_pressure_boundary_conditions_density_h
+
+#include <aspect/fluid_pressure_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+
+
+namespace aspect
+{
+  namespace FluidPressureBoundaryConditions
+  {
+    /**
+     * A class that implements simple fluid pressure boundary conditions
+     * based on the fluid or solid density from the material model.
```

Comment:
I think it would be useful to state the formula by which you approximate the gradient here.